falcon/request.py

```diff
@@ -838,6 +838,38 @@ def get_param_as_list(self, name,
 
         raise HTTPMissingParam(name)
 
+    def get_param_as_date(self, name, format_string="%Y-%m-%d", required=False, store=None):
+        """
+        Args:
+            name (str): Parameter name, case-sensitive (e.g., 'ids').
+            format_string (str): String used to parse the param value into a date.
+                See https://docs.python.org/3.4/library/datetime.html#strftime-and-strptime-behavior
+                (default ``"%Y-%m-%d"``)
+            required (bool, optional): Set to ``True`` to raise
+                ``HTTPBadRequest`` instead of returning ``None`` when the
+                parameter is not found (default ``False``).
+            store (dict, optional): A ``dict``-like object in which to place
+                the value of the param, but only if the param is found (default
+                ``None``).
+        Returns:
+            datetime.date 
+
+        Raises:
+            HTTPBadRequest: A required param is missing from the request.
+            HTTPInvalidParam: A tranform function raised an instance of
+                ``ValueError``.
+        """
+        param_value = self.get_param(name, required=required)
+        try:
+            date = datetime.strptime(param_value, format_string).date()
+        except ValueError:
+            msg = "The date value does not match the required format"
+            raise HTTPInvalidParam(msg, name)
+            
+        if store != None:
```

Comment:
They will both work.  'Is' tests for identity meaning it works in this case
only because None is preinitialized as a constant by python and any
reference to it is a reference to the same object. We really want to test
for equality which is what the '!=' is doing, so I use that because its
more explicit about what I intend. Although they will both return the same
value in this case.

On Mon, Apr 13, 2015, 1:54 PM John Vrbanac <notifications@github.com> wrote:

> In falcon/request.py
> <https://github.com/falconry/falcon/pull/483#discussion_r28261328>:
>
> > +        Returns:
> > +            datetime.date
> > +
> > +        Raises:
> > +            HTTPBadRequest: A required param is missing from the request.
> > +            HTTPInvalidParam: A tranform function raised an instance of
> > +                ``ValueError``.
> > +        """
> > +        param_value = self.get_param(name, required=required)
> > +        try:
> > +            date = datetime.strptime(param_value, format_string).date()
> > +        except ValueError:
> > +            msg = "The date value does not match the required format"
> > +            raise HTTPInvalidParam(msg, name)
> > +
> > +        if store != None:
>
> I think the correct way of expressing this would be:
>
> if store is not None:
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/falconry/falcon/pull/483/files#r28261328>.
>